SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -0,0 +1,1833 @@
+					{
+						// The following check will return false on non-authoritative servers if the Pawn hasn't been received yet.
+						if (APawn* PawnFromPlayerState = PlayerState->GetPawn())
+						{
+							if (PawnFromPlayerState->IsPlayerControlled() && PawnFromPlayerState->HasAuthority())
+							{
+								PawnFromPlayerState->RemoteRole = ROLE_AutonomousProxy;
+							}
+						}
+					}
+
+					if (!bDormantActor)
+					{
+						UpdateShadowData(EntityId);
+					}
+
+					// TODO - Using bActorHadAuthority should be replaced with better tracking system to Actor entity creation [UNR-3960]
+					// When receiving AuthorityGained from SpatialOS, the Actor role will be ROLE_Authority iff this
+					// worker is receiving entity data for the 1st time after spawning the entity. In all other cases,
+					// the Actor role will have been explicitly set to ROLE_SimulatedProxy previously during the
+					// entity creation flow.
+					if (bActorHadAuthority)
+					{
+						Actor->SetActorReady(true);
+					}
+
+					// We still want to call OnAuthorityGained if the Actor migrated to this worker or was loaded from a snapshot.
+					Actor->OnAuthorityGained();
+				}
+				else
+				{
+					UE_LOG(LogActorSystem, Verbose,
+						   TEXT("Received authority over actor %s, with entity id %lld, which has no channel. This means it attempted to "
+								"delete it earlier, when it had no authority. Retrying to delete now."),
+						   *Actor->GetName(), EntityId);
+					NetDriver->Sender->RetireEntity(EntityId, Actor->IsNetStartupActor());
+				}
+			}
+			else if (Authority == WORKER_AUTHORITY_NOT_AUTHORITATIVE)
+			{
+				if (Channel != nullptr)
+				{
+					Channel->bCreatedEntity = false;
+				}
+
+				// With load-balancing enabled, we already set ROLE_SimulatedProxy and trigger OnAuthorityLost when we
+				// set AuthorityIntent to another worker. This conditional exists to dodge calling OnAuthorityLost
+				// twice.
+				if (Actor->Role != ROLE_SimulatedProxy)
+				{
+					Actor->Role = ROLE_SimulatedProxy;
+					Actor->RemoteRole = ROLE_Authority;
+
+					Actor->OnAuthorityLost();
+				}
+			}
+		}
+	}
+	else if (ComponentSetId == SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID)
+	{
+		if (Channel != nullptr)
+		{
+			Channel->ClientProcessOwnershipChange(Authority == WORKER_AUTHORITY_AUTHORITATIVE);
+		}
+
+		// If we are a Pawn or PlayerController, our local role should be ROLE_AutonomousProxy. Otherwise ROLE_SimulatedProxy
+		if (Actor->IsA<APawn>() || Actor->IsA<APlayerController>())
+		{
+			Actor->Role = (Authority == WORKER_AUTHORITY_AUTHORITATIVE) ? ROLE_AutonomousProxy : ROLE_SimulatedProxy;
+		}
+	}
+}
+
+void ActorSystem::ComponentAdded(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId, Schema_ComponentData* Data)
+{
+	if (ComponentId == SpatialConstants::DORMANT_COMPONENT_ID)
+	{
+		HandleDormantComponentAdded(EntityId);
+		return;
+	}
+
+	if (ComponentId < SpatialConstants::STARTING_GENERATED_COMPONENT_ID
```

Comment:
Looks like this replace `MAX_RESERVED_SPATIAL_SYSTEM_COMPONENT_ID`, what was the reasoning? Did we have instances where this was happening, or is this just a cleanup?

Reply:
hmm. I remember rewriting this bit generally, and it was hard to parse. I went with this because I found it easy to read that if it gets past this point, that means it's a) generated and b) not QBI

can change it back if you feel strongly

Reply:
Nah it's fine - was more interested in the change